src/datatrove/pipeline/writers/parquet.py

```diff
@@ -14,24 +14,31 @@ def __init__(
         self,
         output_folder: DataFolderLike,
         output_filename: str = None,
-        compression: str | None = None,
+        compression: Literal["snappy", "gzip", "brotli", "lz4", "zstd"] | None = None,
         adapter: Callable = None,
         batch_size: int = 1000,
         expand_metadata: bool = False,
         max_file_size: int = 5 * 2**30,  # 5GB
     ):
+        # Validate the compression setting
+        if compression not in {"snappy", "gzip", "brotli", "lz4", "zstd", None}:
+            raise ValueError(
+                "Invalid compression type. Allowed types are 'snappy', 'gzip', 'brotli', 'lz4', 'zstd', or None."
+            )
+
         super().__init__(
             output_folder,
             output_filename,
-            compression,
-            adapter,
+            compression=None,  # Ensure superclass initializes without compression
```

Comment:
Set superclass compression parameter to None to leverage pyarrow's native compression functions.